truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/parser/NodeParser.java

```diff
@@ -463,6 +463,11 @@ private NodeData parseNodeData(TypeElement templateType, List<TypeElement> typeH
                 } else if (names.contains(name)) {
                     field.addError(ElementUtils.getAnnotationValue(mirror, "name"), "Duplicate field name '%s'.", name);
                 }
+
+                if (type == null) {
+                    throw new AssertionError("The type specified in the NodeField annotation cannot be found. A possible cause could be a missing import statement for the type.");
```

Comment:
OK. Agreed that AssertionError is probably not the best choice here. Do you have a preference for an existing exception class from JDK or a custom exception class extending RuntimeException?